completions.go

```diff
@@ -0,0 +1,146 @@
+	Stream bool `json:"stream,omitempty"`
+}
+
+// CompletionResponse is the full response from a request to the completions API
+type CompletionResponse struct {
+	ID      string                     `json:"id"`
+	Object  string                     `json:"object"`
+	Created int                        `json:"created"`
+	Model   string                     `json:"model"`
+	Choices []CompletionResponseChoice `json:"choices"`
+	Usage   CompletionResponseUsage    `json:"usage"`
+}
+
+// CompletionResponseChoice is one of the choices returned in the response to the Completions API
+type CompletionResponseChoice struct {
+	Text         string        `json:"text"`
+	Index        int           `json:"index"`
+	LogProbs     LogprobResult `json:"logprobs"`
+	FinishReason string        `json:"finish_reason"`
+}
+
+// LogprobResult represents logprob result of Choice
+type LogprobResult struct {
+	Tokens        []string             `json:"tokens"`
+	TokenLogprobs []float32            `json:"token_logprobs"`
+	TopLogprobs   []map[string]float32 `json:"top_logprobs"`
+	TextOffset    []int                `json:"text_offset"`
+}
+
+// CompletionResponseUsage is the object that returns how many tokens the completion's request used
+type CompletionResponseUsage struct {
+	PromptTokens     int `json:"prompt_tokens"`
+	CompletionTokens int `json:"completion_tokens"`
+	TotalTokens      int `json:"total_tokens"`
+}
+
+// Completion is a single completion.
+func (c *client) Completion(ctx context.Context, request CompletionRequest) (*CompletionResponse, error) {
+	request.Stream = false
+	// Check if the request provides a model
+	if request.Model == "" {
+		request.Model = c.defaultModel
+	}
+	req, err := c.newRequest(ctx, "POST", "/completions", request)
+	if err != nil {
+		return nil, err
+	}
+	resp, err := c.performRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	output := new(CompletionResponse)
+	if err := getResponseObject(resp, output); err != nil {
+		return nil, err
+	}
+	return output, nil
+}
+
+func (c *client) CompletionStream(ctx context.Context, request CompletionRequest, onData func(*CompletionResponse)) error {
+	request.Stream = true
+	req, err := c.newRequest(ctx, "POST", "/completions", request)
+	if err != nil {
+		return err
+	}
+	resp, err := c.performRequest(req)
+	if err != nil {
+		return err
+	}
+
+	reader := bufio.NewReader(resp.Body)
+	defer resp.Body.Close()
+
+	for {
+		line, err := reader.ReadBytes('\n')
+		if err != nil {
+			return err
+		}
+
+		// Trim whitespace
+		line = bytes.TrimSpace(line)
+		// Check if the line has data
+		if !bytes.HasPrefix(line, dataPrefix) {
+			continue
+		}
+		// Trim the data prefix
+		line = bytes.TrimPrefix(line, dataPrefix)
+
+		// Check if the stream is terminated
+		if bytes.HasPrefix(line, streamTerminationPrefix) {
+			break
+		}
+
+		output := new(CompletionResponse)
+		if err := json.Unmarshal(line, output); err != nil {
+			return err
```

Comment:
this used to use a `fmt.Errorf` like the version used in the fine tuning API. we may want to continue doing the previous behavior just in case anyone was expecting it, but also for consistency.